packages/next/src/server/app-render/app-render.tsx

```diff
@@ -545,11 +545,26 @@ async function generateDynamicRSCPayload(
   }
 
   // Otherwise, it's a regular RSC response.
-  return {
+  const baseResponse = {
     b: ctx.sharedContext.buildId,
     f: flightData,
     S: workStore.isStaticGeneration,
   }
+
+  // For runtime prefetches, we encode the stale time and isPartial flag in the response body
+  // rather than relying on response headers. Both of these values will be transformed
+  // by a transform stream before being sent to the client. Rather than initializing with
+  // real values, we use sentinel values that will be replaced by a transform stream after
+  // we've collected the values during render.
+  if (options?.isRuntimePrefetch) {
+    return {
+      ...baseResponse,
+      iP: '__NEXT_POSTPONED_SENTINEL__' as any,
+      st: '__NEXT_STALE_TIME_SENTINEL__' as any,
+    }
+  }
```

Comment:
You can simplify the replacement by making this an array

```suggestion
  if (options?.isRuntimePrefetch) {
    const cryptographicallyRandomNumber = ...
    return {
      ...baseResponse,
      rp: [cryptographicallyRandomNumber]}
  }
```

Then you can do a match on this one number and just replacement it with `<postponed>,<stale>` like `true,10`

your search logic can now drop all the complexity of the earlier or later index. You don't need to worry about JSON formatting.

Then you will need to read `response.rp[0]` for the postponed flag and `repsosne.rp[1]` for the stale time value

and you don't need to have special sentinel values that still need some entropy